app/models/manageiq/providers/ibm_cloud/inventory/collector.rb

```diff
@@ -1,4 +1,25 @@
 class ManageIQ::Providers::IbmCloud::Inventory::Collector < ManageIQ::Providers::Inventory::Collector
   require_nested :PowerVirtualServers
   require_nested :VPC
+
+  private
+
+  def all(api_client, list_method, args, collection_key = nil, opts = {})
+    return enum_for(:all, api_client, list_method, args, collection_key, opts) unless block_given?
+
+    collection_key ||= list_method.to_s.sub("list_", "")
+    result = api_client.send(list_method, *args, opts)
+
+    result.send(collection_key)&.each { |i| yield i }
+
+    next_link = result._next&.href if result.respond_to?(:_next)
+    return unless next_link
+
+    start = parse_next_link(next_link)["start"]&.first
+    all(api_client, list_method, args, collection_key, opts.merge(:start => start))
```

Comment:
we want to avoid recursion only because it hurts the stack depth.  I think we could do all of this with a simple while loop.